source/visionEnhancementProviders/screenCurtain.py

```diff
@@ -66,30 +86,247 @@ class Magnification:
+
+	@classmethod
+	def getTranslatedName(cls) -> str:
+		return screenCurtainTranslatedName
+
+	@classmethod
+	def _get_preInitSettings(cls) -> SupportedSettingType:
+		return [
+			driverHandler.BooleanDriverSetting(
+				"warnOnLoad",
+				warnOnLoadCheckBoxText,
+				defaultVal=True
+			),
+		]
+
+	def _get_supportedSettings(self) -> SupportedSettingType:
+		return super().supportedSettings
+
+
+warnOnLoadText = _(
+	# Translators: A warning shown when activating the screen curtain.
+	# the translation of "Screen Curtain" should match the "translated name"
+	"Enabling Screen Curtain will make the screen of your computer completely black. "
+	"Ensure you will be able to navigate without any use of your screen before continuing. "
+	"\n\n"
+	"Do you wish to continue?"
+)
+
+
+class WarnOnLoadDialog(gui.nvdaControls.MessageDialog):
+
+	showWarningOnLoadCheckBox: wx.CheckBox
+	noButton: wx.Button
+
+	def __init__(
+			self,
+			screenCurtainSettingsStorage: ScreenCurtainSettings,
+			parent,
+			title=_("Warning"),
+			message=warnOnLoadText,
+			dialogType=gui.nvdaControls.MessageDialog.DIALOG_TYPE_WARNING
+	):
+		self._settingsStorage = screenCurtainSettingsStorage
+		super().__init__(parent, title, message, dialogType)
+		self.noButton.SetFocus()
+
+	def _addContents(self, contentsSizer):
+		self.showWarningOnLoadCheckBox: wx.CheckBox = wx.CheckBox(
+			self,
+			label=warnOnLoadCheckBoxText
+		)
+		contentsSizer.addItem(self.showWarningOnLoadCheckBox)
+		self.showWarningOnLoadCheckBox.SetValue(
+			self._settingsStorage.warnOnLoad
+		)
+
+	def _addButtons(self, buttonHelper):
+		yesButton = buttonHelper.addButton(
+			self,
+			id=wx.ID_YES,
+			# Translators: A button in the screen curtain warning dialog which allows the user to
+			# agree to enabling the curtain.
+			label=_("&Yes")
+		)
+		yesButton.Bind(wx.EVT_BUTTON, lambda evt: self._exitDialog(wx.YES))
+
+		noButton: wx.Button = buttonHelper.addButton(
+			self,
+			id=wx.ID_NO,
+			# Translators: A button in the screen curtain warning dialog which allows the user to
+			# disagree to enabling the curtain.
+			label=_("&No")
+		)
+		noButton.SetDefault()
+		noButton.Bind(wx.EVT_BUTTON, lambda evt: self._exitDialog(wx.NO))
+		self.noButton = noButton  # so we can manually set the focus.
+
+	def _exitDialog(self, result: int):
+		"""
+		@param result: either wx.YES or wx.No
+		"""
+		if result == wx.YES:
+			settingsStorage = self._settingsStorage
+			settingsStorage.warnOnLoad = self.showWarningOnLoadCheckBox.IsChecked()
+			settingsStorage._saveSpecificSettings(settingsStorage, settingsStorage.preInitSettings)
+		self.EndModal(result)
+
+	def _onDialogActivated(self, evt):
+		# focus is normally set to the first child, however, we want people to easily be able to cancel this
+		# dialog
+		super()._onDialogActivated(evt)
+		self.noButton.SetFocus()
+
+	def _onShowEvt(self, evt):
+		"""When no other dialogs have been opened first, focus lands in the wrong place (on the checkbox),

```

Comment:
It's not clear to me when this happens. We're now setting focus to no both in _onDialogActivated and in this method. May be this one is obsolete now?

Reply:
This is called when the dialog is "raised" or when focus returns to the dialog. In contrast, `_onDialogActivated` is called when the dialog is first created, on `show` or `showModal`